### PR TITLE
feat(lanelet2_extension)!: introduce API versioning along with format_version

### PR DESCRIPTION
**Pull request by soblin**
## Description

https://github.com/orgs/autowarefoundation/discussions/4740

## Related links

https://github.com/orgs/autowarefoundation/discussions/4740

## Tests performed

https://evaluation.tier4.jp/evaluation/reports/50c7b099-3375-5eb9-8fd5-710fe2011a9c?project_id=prd_jt

## Interface changes

introduce `inline namespace v1`

## Effects on system behavior

none

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [X] I've confirmed the [contribution guidelines].
- [X] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [X] The PR follows the [pull request guidelines].
- [X] The PR has been properly tested.
- [X] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [X] There are no open discussions or they are tracked via tickets.
- [X] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: autoware_lanelet2_extension/lib/query.cpp [hunk 1]
```diff
@@ -50,64 +50,27 @@
 
 using lanelet::utils::to2D;
 
-namespace lanelet::utils
+namespace lanelet::utils::query
```
Comment (youtalk) [inline autoware_lanelet2_extension/lib/query.cpp:53]: This change seems to be a refactoring that is unrelated to the `format_version`. If it's true, I think it should not be included in this PR to clarify the modifications made in this PR.

File: autoware_lanelet2_extension/lib/visualization.cpp [hunk 2]
```diff
@@ -346,158 +346,12 @@ void pushPolygonMarker(
 
 }  // anonymous namespace
 
-namespace lanelet
+namespace lanelet::visualization
```
Comment (youtalk) [inline autoware_lanelet2_extension/lib/visualization.cpp:349]: It's the same as above.

**Review by youtalk — COMMENTED** [review]
The name `namespace v1` did not align with the version in `package.xml` and felt unclear. Instead of forcing a short name, how about using descriptive name like `lanelet_format_v1` or `format_v1`?

**Comment by soblin** [comment]
@youtalk That's right, I'd prefer `format_v1` because it is shorter and we can discriminate it from package version itself !

**Comment by soblin** [comment]
@youtalk After your suggestions I:
- replaced `v1` with `format_v1`
- moved `inline namespace format_v1` to top. So some functions have moved
- only modify the existing functions for the above changes

File: autoware_lanelet2_extension/include/autoware_lanelet2_extension/version.hpp [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+// NOLINTBEGIN(readability-identifier-naming)
+
+namespace lanelet::autoware
+{
+/*
+ * @brief denotes the major format_version
+ */
+enum class Version : int {
+  none = 0,
```
Comment (youtalk) [inline autoware_lanelet2_extension/include/autoware_lanelet2_extension/version.hpp:28]:
```suggestion
  format_none = 0,
```
is better?

**Review by youtalk — APPROVED** [review]
Thank you for your update. It was very easy to review again.
